src/func.c

```diff
@@ -4838,6 +4837,16 @@ void StaticCtorDeclaration::semantic(Scope *sc)
         scope = NULL;
     }
 
+    parent = sc->parent;
+    Dsymbol *p = parent->pastMixin();
+    if (!p->isScopeDsymbol())
+    {
+        ::error(loc, "%sstatic constructor can only be member of module/aggregate/template, not %s %s",
+            (isSharedStaticCtorDeclaration() ? "shared " : ""), p->kind(), p->toChars());
```

Comment:
This is causing building ddmd to fail, because `(? string : string)` will result in a `string` in D which can't be passed to `error`.  Please add casts or a temporary `const char *` variable to make this work again.